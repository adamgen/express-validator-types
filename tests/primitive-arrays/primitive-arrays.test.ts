import { describe, expect, it } from "@jest/globals";
import * as path from "path";
import {
  getTypeAliasTypeStructure,
  getVariableTypeStructure,
} from "../../jest/testFileTypeKeyValues";

import { Project } from "ts-morph";

describe("primitive array schema", () => {
  it("should infer number and string arrays", () => {
    expect(
      getTypeAliasTypeStructure([__dirname, "primitive-arrays.ts"], "myObject")
    ).toEqual({
      string: "string",
      stringArr: "string[]",
      int: "number",
      intArr: "number[]",
    });

    expect(
      getVariableTypeStructure([__dirname, "primitive-arrays.ts"], "schema")
    ).toMatchSnapshot();
  });
  it("should create ts morph resources correctly", () => {
    const project = new Project({});
    const filePath = path.join(__dirname, "primitive-arrays.ts");
    project.addSourceFilesAtPaths(filePath);
    const sourceTsFile = project.getSourceFile(
      path.join(__dirname, "primitive-arrays.ts")
    );

    expect(sourceTsFile.getFilePath()).toBe(
      path.join(__dirname, "primitive-arrays.ts")
    );
  });
});
