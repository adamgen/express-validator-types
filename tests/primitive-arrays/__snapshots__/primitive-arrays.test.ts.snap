// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primitive array schema should infer number and string arrays 1`] = `
{
  "__object."intArr.*"": "{ readonly in: "body"; readonly isInt: true; }",
  "__object."stringArr.*"": "{ readonly in: "body"; readonly isString: true; }",
  "__object.int": "{ readonly in: "body"; readonly isInt: true; }",
  "__object.intArr": "{ readonly in: "body"; readonly isArray: true; readonly optional: true; }",
  "__object.string": "{ readonly in: "body"; readonly isString: true; }",
  "__object.stringArr": "{ readonly in: "body"; readonly isArray: true; readonly optional: true; }",
}
`;
